tck/src/main/java/ee/jakarta/tck/data/framework/read/only/NaturalNumber.java

```diff
@@ -36,9 +36,14 @@ public enum NumberType {
     @jakarta.nosql.Column
     private Short numBitsRequired;
 
+    // Sorting on enum types is vendor-specific in Jakarta Data.
+    // Use numTypeOrdinal for sorting instead.
```

Comment:
If you are already changing this entity why not put ` @Enumerated(STRING)`

for semantic and that sounds better.

Reply:
> If you are already changing this entity why not put ` @Enumerated(STRING)`
> 
> for semantic and that sounds better.

If you like, I can add that too, but it isn't a solution on its own and doesn't fix the problem. Without the other changes I made here, the test would remain broken for providers that are neither JPA nor NoSQL.

Reply:
@njr-11 I know, but once you can change the tests, I recommend using String and updating the tests based on the String instead of int.

Reply:
> @njr-11 I know, but once you can change the tests, I recommend using String and updating the tests based on the String instead of int.

I added Enumerated(STRING) for you under https://github.com/jakartaee/data/pull/831/commits/ba6854459e26d5b50e81a9f93ac98d7b3f23b571
but it is wrong to redo test assertions like you are suggesting in a .0.1 maintenance release.  The scope of a fix in a maintenance should be limited to fixing the problem only, not refactoring tests or switching things around that aren't required to fix the bug.

Reply:
> > @njr-11 I know, but once you can change the tests, I recommend using String and updating the tests based on the String instead of int.
> 
> I added Enumerated(STRING) for you under [ba68544](https://github.com/jakartaee/data/commit/ba6854459e26d5b50e81a9f93ac98d7b3f23b571) but it is wrong to redo test assertions like you are suggesting in a .0.1 maintenance release. The scope of a fix in a maintenance should be limited to fixing the problem only, not refactoring tests or switching things around that aren't required to fix the bug.

I agreed, and that is why I proposed to create those tests as a NoSQL extension:

https://github.com/jakartaee/data/issues/830#issuecomment-2308480914

Reply:
> I agreed, and that is why I proposed to create those tests as a NoSQL extension:
> 
> [#830 (comment)](https://github.com/jakartaee/data/issues/830#issuecomment-2308480914)

Adding some NoSQL specific tests (and also some Jakarta Persistence specific ones) in this are is fine for the 1.1 release.  It would be out of place for the 1.0.1 maintenance release and is not a substitute for fixing the bug that exists in 1.0 / 1.0.1 which this PR addresses.